core/src/test/java/org/wso2/msf4j/util/UtilsTest.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2017, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.msf4j.util;
+
+import org.testng.annotations.Test;
+
+import static org.testng.AssertJUnit.assertEquals;
+
+/**
+ * Tests for utility class.
+ */
+public class UtilsTest {
+
+    @Test
+    public void testNormalizePath() {
+        assertEquals("hello", Utils.normalizePath("hello"));
```

Comment:
Instead of using repetitive assert statements, you can use a data provider.

**Example:**

```java
    @DataProvider(name = "paths")
    public Object[][] paths() {
        return new Object[][]{
                {"/path", "path"},
                {"path/", "path"},
                {"/path/", "path"},
                {"//path", "path"},
                {"///path", "path"},
                {"///path///", "path"},
                {"path1/path2", "path1/path2"},
                {"path1//path2", "path1/path2"},
                {"path1///path2", "path1/path2"},
                {"//path1//path2//", "path1/path2"},
                {"path1/path2/path3/path4", "path1/path2/path3/path4"},
                {"path1//path2//path3//path4", "path1/path2/path3/path4"},
                {"//path1//path2//path3//path4//", "path1/path2/path3/path4"},
                {"", ""},
                {"/", ""},
                {"//", ""},
                {"///", ""}
        };
    }

    /**
     * Test case to test normalization of paths.
     *
     * @param path         test path
     * @param expectedPath expected path after normalising the test path
     */
    @Test(dataProvider = "paths")
    public void normalizePathTest(String path, String expectedPath) {
        String normalizedPath = Utils.normalizePath(path);
        assertEquals(normalizedPath, expectedPath);
    }
```

This will run the above test case multiple times for the given dataset.

WDYT?

Reply:
I chose "hello" because of the "helloWorld" example, but I'm also ok with your approach.  will change and push.